R/board_rsconnect.R

```diff
@@ -71,7 +71,8 @@ board_rsconnect <- function(
     server <- server %||% envvar_get("CONNECT_SERVER") %||% abort("`server` must be supplied")
```

Comment:
I wonder if we should instead use `CONNECT_URL`, which we'd expect to contain the full path, not just the hostname as `SERVER` suggests.

Reply:
Regardless, since I seem to keep messing this up, it also needs a decent comment explaining the problem. And maybe it should perform a simple request (maybe to the version endpoint?) so error early if the server is misconfigured. 

I don't think your approach will work because there's no guarantee that the server will be registered with rsconnect.

Reply:
You're correct - my approach doesn't work for exactly the reason you mention.  I commented on why the current approach doesn't work at https://github.com/rstudio/pins/issues/469#issuecomment-881016721.  I'll edit this PR comment to include the root cause of the problem.

Reply:
I have pushed another attempt at fixing this:
* Without using `rsconnect`
* I inspect the URL at pin write time as well as pin read time
* I created a helper function `rsc_ensure_path_url()` that does the work
*

Reply:
Why can't that just be done on board creation?

Reply:
That's a great question.   I think you're correct that doing this at board creation time is going to be sufficient.

tl'dr:  I'll remove the `rsc_ensure_path_url()` in my next commit.

I understand more about how the package works now, and some of my approach was informed by an intermediate stage where "broken" pins were published on Connect.  It seems that the pin itself (maybe), as well as the local cache (certainly), contains the path to the pin.  This meant that while you have a local cache that contains an incorrect URL, my code was actually necessary.

Now that both I and @juliasilge have cleared our local cache, it seems that the much simpler code base will just work.  In other words, there was this intermediate stage where only a few people would be affected by a broken local cache URL.  Since my testing subjects were in that state, it affected how I thought the package works.

Anyway, inbound commit coming soon.